### DISCUSSION TITLE
field by using react-select with asynchronous function can not persist the value after setState

### DISCUSSION THREAD

**Discussion by 6vvvvvv** [react-hook-form/react-hook-form#8437] [Q&A]
Hi,

I'm now working with the controlled component react-select with asynchronous function and it works fine until I got a display issue like this:
1.when typing words, the dropdown list content will be rendered based my text by calling the api, 
![image](https://user-images.githubusercontent.com/47306004/171648306-6df35576-63d1-4e15-a960-e7e4ed4121c5.png)

2. select an option and the selected option will be diplayed correctly in input field
![image](https://user-images.githubusercontent.com/47306004/171648722-a73ee18b-17e7-4e66-8891-06405d12e5d9.png)

3.since the response consist of not only the address info, but also something like country/ region/ geopointTor, theses related fields in the form will be autocompleted by using setValue
![image](https://user-images.githubusercontent.com/47306004/171649541-1bf5fc1a-d99b-4f83-b080-ea63c2f6d1a6.png)
![image](https://user-images.githubusercontent.com/47306004/171649569-69eaac85-8c18-41c5-9fbe-5191d485b69f.png)

FYI, country / region are using react-select as well but with constant dropdown list content, they are not related to any API call.
geopointTor is a field related to geographic coordinate by using simple input element (it has been registered).

Everything is working as expected until here.

At the moment I'm trying to setValue all these related fields, I also want to refresh the marker position in the map based on the geopointTor by using setState. Here is the data flow and the structure and view of related components
![image](https://user-images.githubusercontent.com/47306004/171653901-d75470d5-6569-4bfb-8cf9-3c4e6cb9001a.png)

View
![image](https://user-images.githubusercontent.com/47306004/171654339-516916e7-06e0-437a-85d5-39bb1dbb4de4.png)
![image](https://user-images.githubusercontent.com/47306004/171654845-531d21c4-27ce-432e-8e58-0e14bf53e321.png)

As you can see in the view, the value of field address will no longer persist while the other fields like country / region / geopointTor stay unchanged.

I've made a test : setState with 2s delay. It turns out that the value of address stays at the beginning but it will disappear after 2s if setState(setMakerPosition) in effect then marker position renewed.

Anyway, can somebody give me an idea of how to persist the address value in the input field in my case. Thanks a lot!

My guess: such a state can be persisted only when the dropdown list content is not empty and there should be an option to be bound. The address list content is based on input text and the API call so it is not determined. That's why the value in react-select with async function can not be kept after setState while the other fields like country(normal react-select) will never be removed even after setState since the country field has a constant list itself?

**Comment by 6vvvvvv** [comment]
Solved, please close this issue, thx!

**Reply by bluebill1049** [reply]
would you be able to share some insight on the solution?
